### PR TITLE
show error bar instead of black bar on invalid channel edit

### PR DESCRIPTION
**Pull request by joshblum**
Screen Shot 2019-06-11 at 4 40 20 PM

Screen Shot 2019-06-11 at 4 40 39 PM

cc @keybase/hotpotatosquad

### CODE REVIEW

File: shared/chat/create-channel/index.desktop.tsx [hunk 1]
```diff
@@ -34,7 +15,7 @@ const CreateChannel = (props: Props) => (
         New chat channel
       </Kb.Text>
     </Kb.Box>
-    {errorHeader(props.errorText)}
+    {props.errorText && <Kb.Banner color="red" text={props.errorText} />}
```
Comment (chrisnojima) [inline shared/chat/create-channel/index.desktop.tsx]: need `!!props.errorText here`

File: shared/chat/create-channel/index.native.tsx [hunk 2]
```diff
@@ -4,34 +4,9 @@ import * as Kb from '../../common-adapters'
...
-        {errorText}
-      </Kb.Text>
-    </Kb.Box>
-  )
-}
-
 const CreateChannel = (props: Props) => (
   <Kb.Box>
-    {errorHeader(props.errorText)}
+    {props.errorText && <Kb.Banner color="red" text={props.errorText} />}
```
Comment (chrisnojima) [inline shared/chat/create-channel/index.native.tsx]: same here

File: shared/chat/manage-channels/edit-channel.tsx [hunk 3]
```diff
@@ -51,36 +47,40 @@ class _EditChannel extends React.Component<Props, State> {
...
-            <Text type="Header" style={{marginBottom: globalMargins.tiny, marginTop: globalMargins.tiny}}>
+            <Kb.Text type="Header" style={{marginBottom: globalMargins.tiny, marginTop: globalMargins.tiny}}>
               {this.props.title}
-            </Text>
+            </Kb.Text>
           )
         )}
-        <Box style={{position: 'relative'}}>
-          <Input
+        {this.props.errorText && <Kb.Banner color="red" text={this.props.errorText} />}
```
Comment (chrisnojima) [inline shared/chat/manage-channels/edit-channel.tsx]: same here

File: shared/chat/manage-channels/index.stories.tsx [hunk 4]
```diff
@@ -122,6 +122,10 @@ const load = () => {
     .add('EditChannel', () => (
       <Box style={toPlatformStyle(editChannelStyle)}>
         <EditChannel
+          errorText={''}
```
Comment (chrisnojima) [inline shared/chat/manage-channels/index.stories.tsx]: dont need curlies

File: shared/chat/manage-channels/index.stories.tsx [hunk 5]
```diff
@@ -139,6 +143,10 @@ const load = () => {
     .add('EditChannel - general', () => (
       <Box style={toPlatformStyle(editChannelStyle)}>
         <EditChannel
+          errorText={''}
```
Comment (chrisnojima) [inline shared/chat/manage-channels/index.stories.tsx]: same

File: shared/chat/manage-channels/index.stories.tsx [hunk 6]
```diff
@@ -156,6 +164,31 @@ const load = () => {
     .add('EditChannel - loading', () => (
       <Box style={toPlatformStyle(editChannelStyle)}>
         <EditChannel
+          errorText={''}
```
Comment (chrisnojima) [inline shared/chat/manage-channels/index.stories.tsx]: same

File: shared/chat/manage-channels/index.stories.tsx [hunk 7]
```diff
@@ -156,6 +164,31 @@ const load = () => {
...
+          onSetChannelCreationError={Sb.action('onSetChannelCreationError')}
+          onSaveSuccess={Sb.action('onSaveSuccess')}
+          waitingOnSave={false}
+          teamname={'stripe.usa'}
+          channelName={''}
```
Comment (chrisnojima) [inline shared/chat/manage-channels/index.stories.tsx]: same

File: shared/chat/manage-channels/index.stories.tsx [hunk 8]
```diff
@@ -156,6 +164,31 @@ const load = () => {
     .add('EditChannel - loading', () => (
       <Box style={toPlatformStyle(editChannelStyle)}>
         <EditChannel
+          errorText={''}
+          onSetChannelCreationError={Sb.action('onSetChannelCreationError')}
+          onSaveSuccess={Sb.action('onSaveSuccess')}
+          waitingOnSave={false}
+          teamname={'stripe.usa'}
```
Comment (chrisnojima) [inline shared/chat/manage-channels/index.stories.tsx]: same

File: shared/chat/manage-channels/index.stories.tsx [hunk 9]
```diff
@@ -156,6 +164,31 @@ const load = () => {
...
+          channelName={''}
+          topic={''}
```
Comment (chrisnojima) [inline shared/chat/manage-channels/index.stories.tsx]: same

File: shared/chat/manage-channels/index.stories.tsx [hunk 10]
```diff
@@ -156,6 +164,31 @@ const load = () => {
...
+          deleteRenameDisabled={false}
+          waitingForGetInfo={true}
+          loadChannelInfo={Sb.action('loadChannelInfo')}
+        />
+      </Box>
+    ))
+    .add('EditChannel - loading with error', () => (
+      <Box style={toPlatformStyle(editChannelStyle)}>
+        <EditChannel
+          errorText={'invalid channel name'}
```
Comment (chrisnojima) [inline shared/chat/manage-channels/index.stories.tsx]: same

**Comment by joshblum** [comment]
@chrisnojima this is ready

**Pull request merged**